### PR TITLE
Trait constructors on scala.js

### PR DESCRIPTION
**Pull request by cquiroz**
I'm trying to get `RangesTests` to run on scala.js but I by default get a Linking error 
```
[error] Referring to non-existent method constructor scala.math.Integral.<init>() [error] called from constructor org.scalajs.testsuite.scalalib.RangesTest$aIsIntegral$1$.<init>(scala.runtime.LazyRef,org.scalajs.testsuite.scalalib.RangesTest)
```
After some discussions in gitter I managed to change it to call the trait constructor but there is an extra step of making the call static and I haven't been able to find it. At the moment my error is as below with a minimal test code to reproduce it
```
[error] Referring to non-existent method static scala.math.PartialOrdering.$$init$(scala.math.PartialOrdering)void
[error]   called from constructor hello.Integral$B$2.<init>(java.lang.Object)
```

I've discussed this with @sjrd in Gitter but haven't found a solution. Hence I'm sending this draft PR to get feedback

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/transform/LinkScala2Impls.scala [hunk 1]
```diff
@@ -78,6 +78,12 @@ class LinkScala2Impls extends MiniPhase with IdentityDenotTransformer { thisPhas
         val impl = implMethod(sel.symbol)
         if (impl.exists) Apply(ref(impl), This(currentClass) :: args).withSpan(app.span)
         else app // could have been an abstract method in a trait linked to from a super constructor
+      case Apply(sel @ Select(Super(a, b), _), args)
```
Comment (cquiroz) [inline compiler/src/dotty/tools/dotc/transform/LinkScala2Impls.scala]: This part should be improved but basically tells to do the transform for constructors even on scala.js

File: compiler/src/dotty/tools/dotc/transform/LinkScala2Impls.scala [hunk 2]
```diff
@@ -88,8 +94,13 @@ class LinkScala2Impls extends MiniPhase with IdentityDenotTransformer { thisPhas
     val implName = ImplMethName(meth.name.asTermName)
     val cls = meth.owner
     if (cls.isAllOf(Scala2xTrait))
-      if (meth.isConstructor)
+      if (meth.isConstructor) {
+        if (ctx.settings.scalajs.value) {
+        // Is this the static call?
         cls.info.decl(nme.TRAIT_CONSTRUCTOR).symbol
```
Comment (cquiroz) [inline compiler/src/dotty/tools/dotc/transform/LinkScala2Impls.scala]: I think this is incorrect but haven't been able to figure out the correct call, any suggestions?

File: sandbox/scalajs/src/Integral.scala [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package hello
+
+// Minimal reproduction
+object Integral {
```
Comment (cquiroz) [inline sandbox/scalajs/src/Integral.scala]: I'm using this for testing but it probably should be removed before merging

**Comment by sjrd** [comment]
This was superseded by #7962 and has become moot since #8652.

**Pull request closed without merging**